examples/nidcpower_source_dc_voltage/_helpers.py

```diff
@@ -231,3 +232,111 @@ def use_simulation_option(default: bool) -> Callable[[F], F]:
+            service_class=nims.session_management.GRPC_SERVICE_CLASS,
+        )
+    )
+
+    with contextlib.ExitStack() as stack:
+        reservation = stack.enter_context(
+            reserve_session(
+                session_management_client,
+                measurement_service.context.pin_map_context,
+                instrument_type_id,
+                timeout=60,
+                pin_names=pin_names,
+            )
+        )
+
+        if len(reservation.session_info) != 1:
+            measurement_service.context.abort(
+                grpc.StatusCode.INVALID_ARGUMENT,
+                f"Unsupported number of sessions: {len(reservation.session_info)}",
+            )
+
+        # Leave session open.
+        session = stack.enter_context(
+            create_session(
+                reservation.session_info[0],
+                instrument_type_module,
+                service_options,
+                measurement_service,
+            )
+        )
+
+    return (session, reservation.session_info)
+
+
+def create_session(
+    session_info: session_management.SessionInformation,
+    instrument_type_module,
+    service_options: Optional[ServiceOptions] = None,
+    measurement_service: Optional[MeasurementService] = None,
+) -> Any:
+    """Create driver session based on the instrument type and reserved session."""
+    session_kwargs = {}
+    # To support teststand_fixture functions since they dont have service options
+    if service_options is None:
+        with GrpcChannelPoolHelper() as grpc_channel_pool:
+            grpc_options = instrument_type_module.GrpcSessionOptions(
+                grpc_channel_pool.get_grpc_device_channel(
+                    instrument_type_module.GRPC_SERVICE_INTERFACE_NAME
+                ),
+                session_name=session_info.session_name,
+                initialization_behavior=instrument_type_module.SessionInitializationBehavior.INITIALIZE_SERVER_SESSION,
+            )
+            return instrument_type_module.Session(
+                resource_name=session_info.resource_name, grpc_options=grpc_options
+            )
+
+    if service_options.use_grpc_device:
+        session_grpc_address = service_options.grpc_device_address
+
+        if not session_grpc_address:
+            session_grpc_channel = measurement_service.get_channel(
+                provided_interface=instrument_type_module.GRPC_SERVICE_INTERFACE_NAME,
+                service_class="ni.measurementlink.v1.grpcdeviceserver",
+            )
+        else:
+            session_grpc_channel = measurement_service.channel_pool.get_channel(
+                target=session_grpc_address
+            )
+        session_kwargs["grpc_options"] = instrument_type_module.GrpcSessionOptions(
+            session_grpc_channel,
+            session_name=session_info.session_name,
+            initialization_behavior=instrument_type_module.SessionInitializationBehavior.AUTO,
+        )
```

Comment:
As I pointed out in a comment on one of my other PRs, it would make sense to extract **this** into a helper method.

Reply:
Also, the `session_grpc_address` variable doesn't add any value. I think it's a remnant of the code before `measurement_service.get_channel` was added.

Reply:
Removed `session_grpc_address`. Refactored the code block to separate get channel method and setting grpc options method.

Reply:
You removed a lot more than that.

This is what removing `session_grpc_address` looks like:

```diff
-       session_grpc_address = service_options.grpc_device_address
-
        if not session_grpc_address:
            session_grpc_channel = measurement_service.get_channel(
                provided_interface=instrument_type_module.GRPC_SERVICE_INTERFACE_NAME,
                service_class="ni.measurementlink.v1.grpcdeviceserver",
            )
        else:
            session_grpc_channel = measurement_service.channel_pool.get_channel(
-               target=session_grpc_address
+               target=service_options.grpc_device_address
            )
        session_kwargs["grpc_options"] = instrument_type_module.GrpcSessionOptions(
            session_grpc_channel,
            session_name=session_info.session_name,
            initialization_behavior=instrument_type_module.SessionInitializationBehavior.AUTO,
        )
```